src/shared_utilities/FileReader.jl

```diff
@@ -154,15 +154,15 @@ end
 
 
 """
-    PrescribedDataTemporal(
+    PrescribedDataTemporal{FT}(
         regrid_dirpath,
         infile_path,
         varname,
         date_ref,
         t_start,
         surface_space;
         mono = true,
-    )
+    ) where {FT <: AbstractFloat}
```

Comment:
I realized that I'm passing `FT` to these external `PrescribedDataTemporal` constructors, but `PrescribedDataTemporal` itself doesn't actually have `FT` as a parametric type, and so doesn't constrain `FT <: AbstractFloat`. Maybe I should add `FT as a type parameter for it? @kmdeck

Reply:
hmmm that makes sense to me! if FT is needed in setting up the data from it or anything, having it handy in the type would be useful. and when we set up the simulation we pick the float type and can set it consistently. we can also possibly enforce that this FT is the same as the model FT...possibly in the model constructor? not sure

Reply:
I feel conflicted... FT would only be used in interpolation, I think ([here](https://github.com/CliMA/ClimaLSM.jl/blob/10e457ad9e6930b90181924f8bddd6b33ae446c5/src/shared_utilities/FileReader.jl#L390-L412)). I think it's fine to continue getting FT from the space as we do now, and maybe simpler